apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java

```diff
@@ -165,12 +166,20 @@ public static AbstractSpan activeSpan() {
         return get().activeSpan();
     }
 
+    /**
+    * Plese use `ContextManager.stopSpan(span)` if could, it's can "fail-fast"
+    */
     public static void stopSpan() {
-        stopSpan(activeSpan());
+        final AbstractTracerContext context = get();
+        stopSpan(context.activeSpan(),context);
     }
 
     public static void stopSpan(AbstractSpan span) {
-        if (get().stopSpan(span)) {
+        stopSpan(span,get());
+    }
+
+    private static void stopSpan(AbstractSpan span,final AbstractTracerContext context) {
```

Comment:
```suggestion
    private static void stopSpan(AbstractSpan span, final AbstractTracerContext context) {
```

Reply:
> @qxo Are you still working on this?

Sorry, still busy...
And I don't familiar with the review workflow, and what should I do next ?